### DISCUSSION TITLE
Subselect and virtual property

### DISCUSSION THREAD

**Discussion by yackinn** [mikro-orm/mikro-orm#3003] [Q&A]
What's the best way to automatically attach a virtual property to each request that counts a relationship of the same entity?

The bookings property shouldn't be loaded eagerly as it has access restrictions based on the user.

```ts
// event.entity.ts

@Formula(...)
@Property({ persist: false })
bookingsCount: number;

@OneToMany(() => Booking, (booking) => booking.event, { nullable: true })
bookings?: Collection<Booking>;
```

**Comment by B4nan** [comment]
If you want dynamic value, that is for example based on a subquery, you can use formulas, note that you don't need `@Property` decorator at all, keep just the `@Formula` one. 

Alternatively you can have a regular (persisted) property and maintain its value. There are several ways you can go about that, e.g. having dedicated `add/remove` methods for the collection that you would use instead of working with the collection instance directly. Or you might want to do it automatically via `onFlush` event.

Btw there is no such thing as nullable collection. If the FK is null, the item is not in the collection, collections are virtual, they can't be nullable.

**Reply by yackinn** [reply]
Thank you.
How are the add/remove methodd implemented to keep track of the count?
Can you post the link from the docs?

Do you mean using a subscriber?
